api/api_public_blockchain.go

```diff
@@ -398,8 +398,23 @@ func (s *PublicBlockChainAPI) EstimateComputationCost(ctx context.Context, args
 	if rpcGasCap := s.b.RPCGasCap(); rpcGasCap != nil {
 		gasCap = rpcGasCap
 	}
-	_, computationCost, err := DoCall(ctx, s.b, args, blockNrOrHash, vm.Config{}, s.b.RPCEVMTimeout(), gasCap)
-	return (hexutil.Uint64)(computationCost), err
+	result, computationCost, err := DoCall(ctx, s.b, args, blockNrOrHash, vm.Config{}, s.b.RPCEVMTimeout(), gasCap)
+	cc := (hexutil.Uint64)(computationCost)
+
+	if err == nil && result.Unwrap() == vm.ErrOpcodeComputationCostLimitReached {
+		block, err := s.b.BlockByNumberOrHash(ctx, blockNrOrHash)
+		if err != nil {
+			return 0, err
+		}
+		rules := s.b.ChainConfig().Rules(block.Number())
```

Comment:
When `blockNrOrHash` is a keyword like `"latest"`, the actual block number used in `DoCall()` can be different from the number used to get `rules`. The node can process more blocks during the execution of `DoCall()`.

It would not be an issue because the computation cost limit will not be changed frequently and this API is just for estimation. So, we can go with this code if providing consistency requires many changes. We can just type a comment, though.

Reply:
For the first paragraph, if a proper block number is required, the user must pass the specific block number. Also, most of other APIs just take and proceed with the string parameters "latest", "pending", and "earlier". Could you give concrete view of your consideration? (why the problem matters only here )